vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/data_transformation.tsx

```diff
@@ -0,0 +1,219 @@
+import React, {useRef, useState} from 'react';
+import {HotTable} from '@handsontable/react';
+import {HyperFormula} from 'hyperformula';
+import EditIcon from '@material-ui/icons/Edit';
+import SaveIcon from '@material-ui/icons/Save';
+import CancelIcon from '@material-ui/icons/Cancel';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import {makeStyles} from '@material-ui/core/styles';
+
+import {joinNesting} from './monoids';
+import {WithInputParams, DataEnvelope} from './input_types';
+import {useSetsDefault} from './useSetsDefault';
+import {some, Maybe} from '../../../../selectors/maybe';
+import {useMemoized} from '../../../../selectors/workflow_selectors';
+
+const useStyles = makeStyles((theme) => ({
+  dialog: {
+    maxWidth: '90vw',
+    maxHeight: '90vh'
+  }
+}));
+
+function DataTransformationModal({
+  data,
+  onChange,
+  onClose
+}: {
+  data: any[][];
+  onChange: (data: Maybe<{[key: string]: any}>) => void;
+  onClose: () => void;
+}) {
+  const classes = useStyles();
+
+  const hotTableComponent = useRef<any>(null);
+  const hyperformulaInstance = HyperFormula.buildEmpty({
```

Comment:
This isn't like a `useState` value?  It should be a unique instance every render?

Reply:
Ah, thanks!